source/tests/system/nitclk_driver_api_tests.cpp

```diff
@@ -0,0 +1,268 @@
+    ResetStubs();
+  }
+
+  virtual ~NiTClkDriverApiTest() {}
+
+  void SetUp() override
+  {
+    initialize_scope_session();
+  }
+
+  void TearDown() override
+  {
+    close_scope_session();
+  }
+
+  void ResetStubs()
+  {
+    channel_ = server_->InProcessChannel(::grpc::ChannelArguments());
+    niscope_stub_ = scope::NiScope::NewStub(channel_);
+    nitclk_stub_ = tclk::NiTClk::NewStub(channel_);
+  }
+
+  std::unique_ptr<scope::NiScope::Stub>& GetScopeStub()
+  {
+    return niscope_stub_;
+  }
+
+  std::unique_ptr<tclk::NiTClk::Stub>& GetTClkStub()
+  {
+      return nitclk_stub_;
+  }
+
+  int GetScopeSessionId()
+  {
+    return scope_session_->id();
+  }
+
+  void initialize_scope_session()
+  {
+    ::grpc::ClientContext context;
+    scope::InitWithOptionsRequest request;
+    request.set_resource_name("FakeDevice");
+    request.set_option_string("Simulate=1, DriverSetup=Model:5164; BoardType:PXIe");
+    request.set_session_name("");
+    request.set_reset_device(false);
+    request.set_id_query(false);
+    scope::InitWithOptionsResponse response;
+
+    ::grpc::Status status = GetScopeStub()->InitWithOptions(&context, request, &response);
+    scope_session_ = std::make_unique<nidevice_grpc::Session>(response.vi());
+
+    ASSERT_TRUE(status.ok());
+    ASSERT_EQ(kScopeDriverApiSuccess, response.status());
+  }
+
+  void close_scope_session()
+  {
+    ::grpc::ClientContext context;
+    scope::CloseRequest request;
+    request.mutable_vi()->set_id(scope_session_->id());
+    scope::CloseResponse response;
+
+    ::grpc::Status status = GetScopeStub()->Close(&context, request, &response);
+
+    EXPECT_TRUE(status.ok());
+    EXPECT_EQ(kScopeDriverApiSuccess, response.status());
+  }
+
+  ViReal64 get_real64_attribute(const char* channel_name, tclk::NiTClkAttributes attribute_id)
+  {
+    ::grpc::ClientContext context;
+    tclk::GetAttributeViReal64Request request;
+    request.mutable_session()->set_id(GetScopeSessionId());
+    request.set_channel_name(channel_name);
+    request.set_attribute_id(attribute_id);
+    tclk::GetAttributeViReal64Response response;
+
+    ::grpc::Status status = GetTClkStub()->GetAttributeViReal64(&context, request, &response);
+
+    EXPECT_TRUE(status.ok());
+    EXPECT_EQ(kTClkDriverApiSuccess, response.status());
+    return response.value();
+  }
+
+  int get_session_id_from_attribute(const char* channel_name, tclk::NiTClkAttributes attribute_id)
+  {
+    ::grpc::ClientContext context;
+    tclk::GetAttributeViSessionRequest request;
+    request.mutable_session()->set_id(GetScopeSessionId());
+    request.set_channel_name(channel_name);
+    request.set_attribute_id(attribute_id);
+    tclk::GetAttributeViSessionResponse response;
+
+    ::grpc::Status status = GetTClkStub()->GetAttributeViSession(&context, request, &response);
+
+    EXPECT_TRUE(status.ok());
+    EXPECT_EQ(kTClkDriverApiSuccess, response.status());
+    return response.mutable_value()->id();
```

Comment:
You shouldn't need to get the `mutable_value` to get the `id` out of it.

Reply:
Changed it to return `response.value().id();`.